File: redome-front/src/app/paciente/notificacao.ts
import { BaseEntidade } from "../shared/base.entidade";
import { Paciente } from './paciente';
import { ConvertUtil } from "app/shared/util/convert.util";
import { UsuarioLogado } from "../shared/dominio/usuario.logado";
import { CategoriaNotificacao } from "app/shared/classes/categoria.notificacao";


/**
 * Classe que representa uma notificação, associada ao paciente.
 * @author Pizão
 */
export class Notificacao extends BaseEntidade {

	private _id: number;
	private _descricao: string;
	private _categoria: CategoriaNotificacao;
	private _rmr: number;
	private _lido: boolean;
	private _dataCriacao: Date;
	private _dataLeitura: Date;	
	private _parceiro: number;
	private _usuario: UsuarioLogado;
	private _aging: string;


	public get id(): number {
		return this._id;
	}

	public set id(value: number) {
		this._id = value;
	}

    /**
     * Getter notificado
     * @return {boolean}
     */
	public get lido(): boolean {
		return this._lido;
	}

    /**
     * Setter notificado
     * @param {boolean} value
     */
	public set lido(value: boolean) {
		this._lido = value;
	}

	public get descricao(): string {
		return this._descricao;
	}

	public set descricao(value: string) {
		this._descricao = value;
	}

	public get rmr(): number {
		return this._rmr;
	}

	public set rmr(value: number) {
		this._rmr = value;
	}

    /**
     * Getter categoria
     * @return {CategoriaNotificacao}
     */
	public get categoria(): CategoriaNotificacao {
		return this._categoria;
	}

    /**
     * Setter categoria
     * @param {CategoriaNotificacao} value
     */
	public set categoria(value: CategoriaNotificacao) {
		this._categoria = value;
	}

    /**
     * Getter dataCriacao
     * @return {Date}
     */
	public get dataCriacao(): Date {
		return this._dataCriacao;
	}

    /**
     * Setter dataCriacao
     * @param {Date} value
     */
	public set dataCriacao(value: Date) {
		this._dataCriacao = value;
	}

    /**
     * Getter dataLeitura
     * @return {Date}
     */
	public get dataLeitura(): Date {
		return this._dataLeitura;
	}

    /**
     * Setter dataLeitura
     * @param {Date} value
     */
	public set dataLeitura(value: Date) {
		this._dataLeitura = value;
	}

    /**
     * Getter parceiro
     * @return {number}
     */
	public get parceiro(): number {
		return this._parceiro;
	}

    /**
     * Setter parceiro
     * @param {number} value
     */
	public set parceiro(value: number) {
		this._parceiro = value;
	}

    /**
     * Getter usuario
     * @return {UsuarioLogado}
     */
	public get usuario(): UsuarioLogado {
		return this._usuario;
	}

    /**
     * Setter usuario
     * @param {UsuarioLogado} value
     */
	public set usuario(value: UsuarioLogado) {
		this._usuario = value;
	}

    /**
     * Getter aging
     * @return {string}
     */
	public get aging(): string {
		return this._aging;
	}

    /**
     * Setter aging
     * @param {string} value
     */
	public set aging(value: string) {
		this._aging = value;
	}



	public jsonToEntity(res: any): Notificacao {

		if (res.categoria) {
			this.categoria = new CategoriaNotificacao().jsonToEntity(res.categoria);
		}

		if (res.usuario) {
			this.usuario = new UsuarioLogado().jsonToEntity(res.usuario);
		}

		this.id = ConvertUtil.parseJsonParaAtributos(res.id, new Number());
		this.descricao = ConvertUtil.parseJsonParaAtributos(res.descricao, new String());
		this.rmr = ConvertUtil.parseJsonParaAtributos(res.rmr, new Number());
		this.lido = ConvertUtil.parseJsonParaAtributos(res.lido, new Boolean());
		this.dataCriacao = ConvertUtil.parseJsonParaAtributos(res.dataCriacao, new Date());
		this.dataLeitura = ConvertUtil.parseJsonParaAtributos(res.dataLeitura, new Date());
		this.parceiro = ConvertUtil.parseJsonParaAtributos(res.parceiro, new Number());
		this.aging = ConvertUtil.parseJsonParaAtributos(res.aging, new String());

		return this;
    }
}

File: redome-notificacao/src/main/java/br/org/cancer/redome/notificacao/model/Notificacao.java
package br.org.cancer.redome.notificacao.model;

import java.io.Serializable;
import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Transient;
import javax.validation.constraints.NotNull;

import org.hibernate.annotations.CreationTimestamp;

import br.org.cancer.redome.notificacao.dto.NotificacaoDTO;
import lombok.Builder;
import lombok.Builder.Default;
import lombok.Data;

/**
 * Classe que define os atributos contidos na entidade
 * que representa uma notificação associado a um determinado
 * paciente enviado para um determinado usuário.
 * 
 * @author ergomes
 * 
 */
@Entity
@SequenceGenerator(name = "SQ_NOTI_ID", sequenceName = "SQ_NOTI_ID", allocationSize = 1)
@Table(name = "NOTIFICACAO")
@Data
@Builder
public class Notificacao implements Serializable {
	private static final long serialVersionUID = -6115279067378891892L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SQ_NOTI_ID")
	@Column(name = "NOTI_ID")
	private Long id;

	@Column(name = "NOTI_TX_DESCRICAO")
	private String descricao;

	@OneToOne
	@JoinColumn(name = "CANO_ID")
	private CategoriaNotificacao categoria;

	@Column(name = "PACI_NR_RMR")
	private Long rmr;

	@Column(name = "NOTI_IN_LIDO")
	@Default
	private Boolean lido = false;
	
	/**
	 * Data de criação da notificação.
	 */
	@Column(name = "NOTI_DT_CRIACAO")
	@NotNull
	@CreationTimestamp
	private LocalDateTime dataCriacao;

	/**
	 * Data de leitura da notificação.
	 */
	@Column(name = "NOTI_DT_LIDO")
	protected LocalDateTime dataLeitura;	
	
	/**
	 * Parceiro para o usuário tem relação.
	 */
	@Column(name = "NOTI_ID_PARCEIRO")
	private Long parceiro;

	/**
	 * Usuário para o qual foi criada a notificação.
	 */
	@Transient
	@NotNull
	private Long usuario;
	
	public Notificacao() {
		this.dataCriacao = LocalDateTime.now();
	}

	/**
	 * @param id
	 * @param descricao
	 * @param categoria
	 * @param paciente
	 * @param lido
	 * @param dataCriacao
	 * @param dataLeitura
	 * @param parceiro
	 * @param usuario
	 */
	public Notificacao(Long id, String descricao, CategoriaNotificacao categoria, Long rmr, Boolean lido,
			LocalDateTime dataCriacao, LocalDateTime dataLeitura, Long parceiro, Long usuario) {
		super();
		this.id = id;
		this.descricao = descricao;
		this.categoria = categoria;
		this.rmr = rmr;
		this.lido = lido;
		this.dataCriacao = dataCriacao;
		this.dataLeitura = dataLeitura;
		this.parceiro = parceiro;
		this.usuario = usuario;
	}
	
	public static Notificacao parse(NotificacaoDTO notificacaoDto) {
		
		Notificacao notificacaoRetorno = new Notificacao();
		
		notificacaoRetorno.setId(notificacaoDto.getId());
		notificacaoRetorno.setDescricao(notificacaoDto.getDescricao());
		notificacaoRetorno.setCategoria(new CategoriaNotificacao(notificacaoDto.getCategoriaId()));
		notificacaoRetorno.setRmr(notificacaoDto.getRmr());
		notificacaoRetorno.setLido(notificacaoDto.getLido());
		notificacaoRetorno.setDataCriacao(notificacaoDto.getDataCriacao());
		notificacaoRetorno.setDataLeitura(notificacaoDto.getDataLeitura());
		notificacaoRetorno.setParceiro(notificacaoDto.getParceiro());
		//notificacaoRetorno.setUsuario(notificacaoDto.getUsuarioId());
		
		return notificacaoRetorno;
	}

}